src/cli/api/local_api.py

```diff
@@ -188,11 +200,49 @@ def get_network_data(self, network_name: str)-> assetNetworkType:
             The complete network information with channels & nodes
 
         """
-        return {}
+
+        all_network_nodes = utils.get_network_nodes()
+        channels_list = []
+        nodes_list = []
+
+        network_slug = utils.get_network_slug(network_name)
+
+        if network_slug:
+            channels = utils.get_channels_for_network(network_slug=network_slug)
+            if channels:
+                for channel_slug in channels:
+                    channel_info = utils.get_channel_info(channel_slug=channel_slug)
+                    channels_list.append(channel_info)
+
+            if network_slug in all_network_nodes:
+                for node_slug in all_network_nodes[network_slug]:
+                    nodes_list.append(utils.get_node_info(node_slug=node_slug))
+
+        return {
+            "name": network_name,
+            "slug": network_slug,
+            "channels": channels_list,
+            "nodes": nodes_list,
+        }
 
     def create_experiment(
         self, name: str, app_config: AppConfigType, asset_network: assetNetworkType, path: Path, application: str
     ) -> Tuple[bool, str]:
+        """
+        Create experiment.json with meta and asset information
+
+        Args:
+            name: Name of the directory where experiment.json will be created
+            app_config: A dictionary containing application configuration information
+            asset_network: Filled Network parameters with default values
+            path: Location where experiment directory needs to be created
+            application: Name of the application for which to create the experiment
+
+        Returns:
+            (True, Success) if experiment.json was successfully created
```

Comment:
create application throws an exception when application already exists. Here a return value is returned. This must be consistent for each method.

Reply:
We do a lot of work before testing experiment already exists, maybe this check can be done earlier?